### ISSUE TITLE
Lookahead Optimizer: k steps forward, 1 step back

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#43]
## [Lookahead Optimizer: k steps forward, 1 step back](https://papers.nips.cc/paper/9155-lookahead-optimizer-k-steps-forward-1-step-back)
*The vast majority of successful deep neural networks are trained using variants of stochastic gradient descent (SGD) algorithms. Recent attempts to improve SGD can be broadly categorized into two approaches: (1) adaptive learning rate schemes, such as AdaGrad and Adam, and (2) accelerated schemes, such as heavy-ball and Nesterov momentum. In this paper, we propose a new optimization algorithm, Lookahead, that is orthogonal to these previous approaches and iteratively updates two sets of weights. Intuitively, the algorithm chooses a search direction by looking ahead at the sequence of fast weights generated by another optimizer. We show that Lookahead improves the learning stability and lowers the variance of its inner optimizer with negligible computation and memory cost. We empirically demonstrate Lookahead can significantly improve the performance of SGD and Adam, even with their default hyperparameter settings on ImageNet, CIFAR-10/100, neural machine translation, and Penn Treebank.*

### Cite this paper
```bibtex
@incollection{NIPS2019_9155,
  title = {Lookahead Optimizer: k steps forward, 1 step back},
  author = {Zhang, Michael and Lucas, James and Ba, Jimmy and Hinton, Geoffrey E},
  booktitle = {Advances in Neural Information Processing Systems 32},
  editor = {H. Wallach and H. Larochelle and A. Beygelzimer and F. d\textquotesingle Alch\'{e}-Buc and E. Fox and R. Garnett},
  pages = {9597--9608},
  year = {2019},
  publisher = {Curran Associates, Inc.},
  url = {http://papers.nips.cc/paper/9155-lookahead-optimizer-k-steps-forward-1-step-back.pdf}
}
```
